tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -147,131 +164,79 @@ Note that:
     end
 end;
 ```
 
-The function `pPCA(Array)` is constructed using multi-dispatch.
-The function accepts
+The model function `pPCA()` accepts:
 
- 1. an Array $x$ (with dimension no. of instances times no. of features) ;
- 2. optionally: an integer $k$ which is the dimension of the latent space (the space the original feature matrix is projected onto)
-    as input argument and defines the probabilistic model for p-PCA.
+ 1. an data array $\mathbf{X}$ (with dimension no. of features x no. of instances) ;
+ 2. an integer $k$ which is the dimension of the latent space (the space the original feature matrix is projected onto).
 
 Specifically:
 
  1. it first extracts the dimension $D$ and number of instances $N$ of the input matrix;
- 2. then draw $N$ samples of the latent variable $z$ with dimension $k$ from an MND;
- 3. then draw samples of each entries of the projection matrix $w$ (with dimension $D \times k$), from a standard normal;
- 4. then we assume uniform offset for each dimension $d \in D$, and draw samples from an MND for each dimension;
- 5. Finally, we iterate through each dimension in $D$, and define an MND (sampling distribution, likelihood), assuming all instances in that dimension follow an MND with mean $w \times z + mu_d$, for all instances in that dimension.
+ 2. draw samples of each entries of the projection matrix $\mathbf{W}$ from a standard normal;
+ 3. draw samples of the latent variable $\mathbf{Z}_{k \times N}$ from an MND;
+ 4. draw samples of the offset $\boldsymbol{\mu}$ from an MND, assuming uniform offset for all instances;
+ 5. Finally, we iterate through each gene dimension in $\mathbf{X}$, and define an MND for the sampling distribution (i.e. likelihood).
 
 #### Step 4: Sampling-based inference of the pPCA model
 
-Here we aim to perform MCMC sampling to infer the projection matrix $w_{D \times k}$, the latent variable matrix $z_{k \times N}$, and the offsets $mu_{D \times 1}$.
+Here we aim to perform MCMC sampling to infer the projection matrix $\mathbf{W}_{D \times k}$, the latent variable matrix $\mathbf{Z}_{k \times N}$, and the offsets $\boldsymbol{\mu}_{N \times 1}$.
 
-We run the inference using the NUTS sampler, of which the chain length is set to be 500. By default, the NUTS sampler samples 1 chain.
+We run the inference using the NUTS sampler, of which the chain length is set to be 500, target accept ratio 0.65 and initial stepsize 0.1. By default, the NUTS sampler samples 1 chain.
 You are free to try [different samplers](https://turing.ml/stable/docs/library/#samplers).
 
 ```julia
 k = 2 # k is the dimension of the projected space, i.e. the number of principal components/axes of choice
-ppca = pPCA(expression_matrix, k) # instantiate the probabilistic model
-chain_ppca = sample(ppca, NUTS(), 500);
+ppca = pPCA(Array(mat_exp), k) # ppca = pPCA_1(Array(df_exp), k), instantiate the probabilistic model
+chain_ppca = sample(ppca, NUTS(0.65; init_ϵ=0.1), 500);
```

Comment:
Is it necessary to set options for `NUTS` explicitly? Can't we just use the default?

Reply:
True, it is redundant to instruct intial stepsize and accept rate - just removed them.